src/plugins/proxy/src/plugin.cpp

```diff
@@ -0,0 +1,572 @@
+    return config.find(ov::device::id.name()) != config.end();
+}
+
+size_t get_device_from_config(const ov::AnyMap& config) {
+    if (is_device_in_config(config))
+        return config.at(ov::device::id.name()).as<size_t>();
+    return 0;
+}
+
+ov::AnyMap remove_device_properties(ov::AnyMap& config, const std::vector<std::string>& devices) {
+    ov::AnyMap result;
+    std::unordered_set<std::string> devs;
+    for (const auto& dev : devices)
+        devs.insert(dev);
+
+    for (const auto& it : config) {
+        auto subprop_device_name_pos = it.first.find(ov::device::properties.name() + std::string("_"));
+        if (subprop_device_name_pos == std::string::npos)
+            continue;
+        auto subprop_device_name =
+            it.first.substr(subprop_device_name_pos + std::strlen(ov::device::properties.name()) + 1);
+        ov::DeviceIDParser parser(subprop_device_name);
+        if (devs.find(subprop_device_name) != devs.end() || devs.find(parser.get_device_name()) != devs.end()) {
+            // It is a device property
+            result[subprop_device_name] = it.second;
+        }
+    }
+
+    // Remove device properties from config
+    for (const auto& it : result) {
+        auto c_it = config.find(ov::device::properties.name() + std::string("_") + it.first);
+        if (c_it != config.end())
+            config.erase(c_it);
+    }
+    return result;
+}
+
+ov::AnyMap remove_proxy_properties(ov::AnyMap& config, bool rem_device_properties = false) {
+    ov::AnyMap dev_properties;
+    auto it = config.find(ov::device::id.name());
+    if (it != config.end())
+        config.erase(it);
+    it = config.find(ov::device::priorities.name());
+    if (it != config.end()) {
+        if (rem_device_properties)
+            dev_properties = remove_device_properties(config, it->second.as<std::vector<std::string>>());
+        config.erase(it);
+    }
+    it = config.find(ov::proxy::alias_for.name());
+    if (it != config.end())
+        config.erase(it);
+    it = config.find(ov::proxy::device_priorities.name());
+    if (it != config.end())
+        config.erase(it);
+    return dev_properties;
+}
+
+}  // namespace
+
+ov::proxy::Plugin::Plugin() {
+    // Create global config
+    m_configs[""] = {};
+}
+ov::proxy::Plugin::~Plugin() = default;
+
+ov::SupportedOpsMap ov::proxy::Plugin::query_model(const std::shared_ptr<const ov::Model>& model,
+                                                   const ov::AnyMap& properties) const {
+    size_t num_devices = get_hidden_devices().size();
+    // Recall for HW device
+    auto dev_id = get_device_from_config(properties);
+    auto config_copy = properties;
+    remove_proxy_properties(config_copy);
+    auto res = get_core()->query_model(model, get_fallback_device(dev_id), config_copy);
+    // Replace hidden device name
+    for (auto&& it : res) {
+        it.second = get_device_name();
+        if (num_devices > 1)
+            it.second += "." + std::to_string(dev_id);
```

Comment:
I think you can always mark with device id

Reply:
In case if plugin detects only one device plugins return the device name without index (for example CPU returns CPU instead of CPU.0)

Reply:
dev_id can be empty string and even CPU supports it now